### ISSUE TITLE
The unless requisite stops at first successful command

### ISSUE THREAD

**Issue by grobinson-blockchain** [saltstack/salt#35384]
I'm having trouble with the `unless` requisite where it checks the first command but no others:

    nginx-build:
      cmd.run:
        - name: ./configure --prefix=/etc/nginx --sbin-path=/usr/sbin/nginx && make
        - cwd: /tmp/nginx-{{ nginx.version }}
        - require:
          - archive: nginx-download
        - unless:
          - /usr/sbin/nginx -v 2>&1 | grep {{ nginx.version }}
          - test "hello" = "world"

The docs for `unless` state that:

> The unless requisite specifies that a state should only run when any of the specified commands return False. The unless requisite operates as NAND and is useful in giving more granular control over when a state should execute.

However, if I pass `--log-level=debug` I see salt execute the first command (which returns True) but never the last command:

    [INFO    ] Running state [./configure --prefix=/etc/nginx --sbin-path=/usr/sbin/nginx --conf-path=/etc/nginx/nginx.conf  && make] at time 13:20:59.214227
    [INFO    ] Executing state cmd.run for ./configure --prefix=/etc/nginx --sbin-path=/usr/sbin/nginx --conf-path=/etc/nginx/nginx.conf  && make
    [INFO    ] Executing command '/usr/sbin/nginx -v 2>&1 | grep 1.11.3' in directory '/tmp/nginx-1.11.3'
    [DEBUG   ] output: nginx version: nginx/1.11.3
    [DEBUG   ] Last command return code: [0]
    [INFO    ] unless execution succeeded

### Versions Report
salt-call 2016.3.2 (Boron)

**Comment by Ch3LL** [comment]
@grobinson-blockchain I think I have tracked this down to only being broken with `cmd.run`. 

If you run this state:

```
vim:
  pkg.installed:
    - unless:
      - rpm -q vim-enhanced
      - ls /usr/bin/vim
```

You will see both commadns in the debug output like so:

```
[INFO    ] Executing command 'rpm -q vim-enhanced' in directory '/root'
[DEBUG   ] output: vim-enhanced-7.4.1868-1.fc22.x86_64
[DEBUG   ] Last command return code: 0
[INFO    ] Executing command 'ls /usr/bin/vim' in directory '/root'
[DEBUG   ] output: /usr/bin/vim
[DEBUG   ] Last command return code: 0
[INFO    ] unless execution succeeded
```

But if you change the state file to cmd.run:

```
echo "hello":
  cmd.run:
    - unless:
      - rpm -q vim-enhanced
      - ls /usr/bin/vim
```

I only see the first comman run:

```
[INFO    ] Executing command 'rpm -q vim-enhanced' in directory '/root'
[DEBUG   ] output: vim-enhanced-7.4.1868-1.fc22.x86_64
[DEBUG   ] Last command return code: [0]
[INFO    ] unless execution succeeded
```

I even tested this in 2015.5.10 and saw the same behavior. We will need to get this fixed up thanks.

**Comment by rallytime** [comment]
This has been fixed by #35545. Thanks to @Ch3LL for the great investigation work to trim down the issue to just the cmd state. I tested the fix with the following state:
```
# cat /srv/salt/unless-test.sls
echo "hello":
  cmd.run:
    - unless:
      - /bin/true
      - /bin/false
```
Which, before the fix, exhibits the broken behavior:
```
# salt-call --local state.sls unless-test
local:
----------
          ID: echo "hello"
    Function: cmd.run
      Result: True
     Comment: unless execution succeeded
     Started: 16:22:01.132921
    Duration: 26.237 ms
     Changes:

Summary for local
------------
Succeeded: 1
Failed:    0
------------
Total states run:     1
Total run time:  26.237 ms
```
But after the fix, it works as expected:
```
# salt-call --local state.sls unless-test
local:
----------
          ID: echo "hello"
    Function: cmd.run
      Result: True
     Comment: Command "echo "hello"" run
     Started: 16:22:09.590493
    Duration: 88.452 ms
     Changes:
              ----------
              pid:
                  18850
              retcode:
                  0
              stderr:
              stdout:
                  hello

Summary for local
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time:  88.452 ms
```

**Comment by rallytime** [comment]
Regression test written in #35569.

**Comment by rallytime** [comment]
This fix will be available in the next release(s) of Salt (2015.8.12 and 2016.3.3). Thanks all!
